### PR TITLE
Changes for the uniqueness of org code and name for serviced organisation

### PR DESCRIPTION
**Pull request by NitishAroraKPMG**

### CODE REVIEW

File: transit_odp/timetables/loaders.py [hunk 1]
```diff
@@ -272,10 +276,12 @@ def load_serviced_organisation(self):
             df_merge_db_and_file_inputs = pd.merge(
                 df_serviced_organisations,
                 df_merged_serviced_orgs,
-                on="serviced_org_ref",
+                on=["serviced_org_ref", "name"],
                 how="inner",
                 suffixes=["_file", "_db"],
             )
+            print("df_merge_db_and_file_inputs")
```
Comment (alvak1) [inline transit_odp/timetables/loaders.py:283]: Remove print statments

File: transit_odp/timetables/loaders.py [hunk 2]
```diff
@@ -272,10 +276,12 @@ def load_serviced_organisation(self):
...
+            print(df_merge_db_and_file_inputs.columns)
```
Comment (alvak1) [inline transit_odp/timetables/loaders.py:284]: Remove print statments

File: transit_odp/timetables/loaders.py [hunk 3]
```diff
@@ -231,8 +231,12 @@ def load_serviced_organisation(self):
 
             existing_serviced_orgs = ServicedOrganisations.objects.all()
             existing_serviced_orgs_list = existing_serviced_orgs.values_list(
-                "organisation_code", flat=True
+                "name", "organisation_code"
             )
+            existing_serviced_orgs_list = [
+                "".join(serviced_org) for serviced_org in existing_serviced_orgs_list
+            ]
+            print(f"existing_serviced_orgs_list: {existing_serviced_orgs_list}")
```
Comment (alvak1) [inline transit_odp/timetables/loaders.py:239]: Remove print statments

**Review by alvak1 — COMMENTED** [review]
Can we add a test case to verify the bug fix, that is explicitly check that serviced org with same org code and same name with different case is stored.

File: transit_odp/pipelines/pipelines/dataset_etl/utils/dataframes.py [hunk 4]
```diff
@@ -213,11 +214,17 @@ def df_to_flexible_service_operation_period(
 
 
 def df_to_serviced_organisations(
-    df: pd.DataFrame, existing_serviced_orgs
+    df: pd.DataFrame, existing_serviced_orgs_list: List[str]
```
Comment (alvak1) [inline transit_odp/pipelines/pipelines/dataset_etl/utils/dataframes.py:216]: Add docstring

File: transit_odp/timetables/loaders.py [hunk 5]
```diff
@@ -231,8 +231,12 @@ def load_serviced_organisation(self):
 
             existing_serviced_orgs = ServicedOrganisations.objects.all()
             existing_serviced_orgs_list = existing_serviced_orgs.values_list(
-                "organisation_code", flat=True
+                "name", "organisation_code"
```
Comment (alvak1) [inline transit_odp/timetables/loaders.py:234]: Add docstring

**Review by alvak1 — COMMENTED** [review]
Add docstrings where required

File: transit_odp/transmodel/migrations/0016_alter_servicedorganisations_organisation_code_and_more.py [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+        ("transmodel", "0015_vehiclejourney_service_pattern_and_more"),
+    ]
+
+    operations = [
+        migrations.AlterField(
+            model_name="servicedorganisations",
+            name="organisation_code",
+            field=models.CharField(blank=True, max_length=255, null=True),
+        ),
+        migrations.AlterUniqueTogether(
```
[the remark this answers is not in the document]
Comment (NitishAroraKPMG) [reply transit_odp/transmodel/migrations/0016_alter_servicedorganisations_organisation_code_and_more.py]: yes, it was generated using makemigrations and later on the uniqueness was changed, hence pushed the new file.

File: transit_odp/pipelines/tests/test_dataset_etl/test_etl_serviced_organisations.py [hunk 7]
```diff
@@ -77,9 +77,39 @@ def test_extract(self):
...
+                    "file_id": file_id,
+                    "serviced_org_ref": "NYCCSC",
+                    "name": "NYCC SCHOOLS",
+                    "start_date": "2023-06-04",
+                    "end_date": "2023-07-25",
+                },
             ]
         ).set_index("file_id")
 
+        print(extracted.serviced_organisations)
```
Comment (alvak1) [inline transit_odp/pipelines/tests/test_dataset_etl/test_etl_serviced_organisations.py:111]: Remove the print statement

**Review by alvak1 — APPROVED** [review]
Approved

**Pull request merged**